plugin/session_buffer.py

```diff
@@ -520,18 +525,17 @@ def _on_document_diagnostic_async(self, version: int, response: DocumentDiagnost
         self._document_diagnostic_pending_request = None
         self._if_view_unchanged(self._apply_document_diagnostic_async, version)(response)
 
-    def _apply_document_diagnostic_async(
-        self, view: sublime.View | None, response: DocumentDiagnosticReport
-    ) -> None:
+    def _apply_document_diagnostic_async(self, view: sublime.View | None, response: DocumentDiagnosticReport) -> None:
         self.session.diagnostics_result_ids[self._last_known_uri] = response.get('resultId')
         if is_full_document_diagnostic_report(response):
             self.session.m_textDocument_publishDiagnostics(
                 {'uri': self._last_known_uri, 'diagnostics': response['items']})
-        for uri, diagnostic_report in response.get('relatedDocuments', {}):
-            sb = self.session.get_session_buffer_for_uri_async(uri)
-            if sb:
-                cast(SessionBuffer, sb)._apply_document_diagnostic_async(
-                    None, cast(DocumentDiagnosticReport, diagnostic_report))
+        if 'relatedDocuments' in response:
+            for uri, diagnostic_report in response['relatedDocuments'].items():
+                sb = self.session.get_session_buffer_for_uri_async(uri)
+                if sb:
+                    cast(SessionBuffer, sb)._apply_document_diagnostic_async(
+                        None, cast(DocumentDiagnosticReport, diagnostic_report))
```

Comment:
This code was just plain wrong before (missing `.items()`) and would creash on non-empty `relatedDocuments`.